it/src/main/java/com/google/cloud/teleport/it/bigquery/DefaultBigQueryResourceManager.java

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.bigquery;
+
+import static com.google.cloud.teleport.it.bigquery.BigQueryResourceManagerUtils.checkValidTableId;
+
+import com.google.api.gax.paging.Page;
+import com.google.auth.Credentials;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryError;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.DatasetInfo;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.InsertAllRequest.RowToInsert;
+import com.google.cloud.bigquery.InsertAllResponse;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardTableDefinition;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableDefinition;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableInfo;
+import com.google.cloud.bigquery.TableResult;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default class for implementation of {@link BigQueryResourceManager} interface.
+ *
+ * <p>The class supports one dataset, and multiple tables per dataset object.
+ *
+ * <p>The class is thread-safe.
+ */
+public final class DefaultBigQueryResourceManager implements BigQueryResourceManager {
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultBigQueryResourceManager.class);
+  private static final String DEFAULT_DATASET_REGION = "us-central1";
```

Comment:
Is there a reason why we have a default region and its not passed in as a builder parameter?

Reply:
That is just a macro constant for defining a region for the dataset that gets created by default when createTable() is called without a previous call to createDataset(). I don't see a reason why the builder would need that value unless we want to make the dataset region configurable which can be achieved by calling the createDataset() method.

Reply:
I think Spanner Resource Manager takes the region as a builder parameter. Using default also makes sense.